### PR TITLE
Revert "Revert "Custom Text for Case List Empty State""

### PR DESCRIPTION
**Pull request by Jtang-1**
## Product Description
Same as described in https://github.com/dimagi/commcare-hq/pull/32491

## Technical Summary
Review commit by commit (many of the line additions are for one commit to update xml files for tests).
Same as described in https://github.com/dimagi/commcare-hq/pull/32491:
Related to Formplayer PR: https://github.com/dimagi/formplayer/pull/1309
Commcare-Core PR: https://github.com/dimagi/commcare-core/pull/1209

New to this PR: 
[USH-2869](https://dimagi-dev.atlassian.net/browse/USH-2869
) Feature UI on app builder, and changes to suite.xml, app_string, and bulk translation are gated behind CC version 2.54 since these changes are dependent on updated commcare-core code.
Associated Formplayer update: https://github.com/dimagi/formplayer/pull/1320

## Feature Flag
Same as described in https://github.com/dimagi/commcare-hq/pull/32491:

### Safety story
Will go through QA for mobile regression testing.

### Automated test coverage
Tests added in PR - covers app string creation, xml creation, and bulk app translation.

### QA Plan
[QA-4760](https://dimagi-dev.atlassian.net/browse/QA-4760)

- [X] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [X] Risk label is set correctly
- [X] The set of people pinged as reviewers is appropriate for the level of risk of the change

[USH-2869]: https://dimagi-dev.atlassian.net/browse/USH-2869?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[QA-4760]: https://dimagi-dev.atlassian.net/browse/QA-4760?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by Jtang-1** [comment]
This wasn't mentioned previously but I've been thinking about it and not certain if it's an issue so wanted to flag it in case - 

In respect to translation backwards compatibility - apps that haven't been rebuilt won't have the new node in the suite file so will pull a hardcoded default value from [formplayer](https://github.com/dimagi/formplayer/blob/94c0b3a3b314a6cd2ed5c08b5b051806f83f7459/src/main/java/org/commcare/formplayer/beans/menus/EntityListResponse.java#L529). Which is then pulled into web apps through [JS](https://github.com/dimagi/commcare-hq/blob/3b522967c301d54c5e13f7620ed8846b6aad3861/corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views.js#L478) and [template](https://github.com/dimagi/commcare-hq/blob/3b522967c301d54c5e13f7620ed8846b6aad3861/corehq/apps/cloudcare/templates/formplayer/case_list.html#L47) . I'm not sure if this will cause users who previously saw this as a translated text to now only see it in english.

If so, a solution I have is to pass back nothing from formplayer if there's no customized text. And on the template, check if `noItemsText` is empty, and if so, display `{% trans "List is empty." %}`. I wasn't able to find a pattern we use to translate default text passed through a variable (even though i'm sure we do it).

File: corehq/apps/app_manager/feature_support.py [hunk 1]
```diff
@@ -208,3 +208,10 @@ def ush_case_claim_2_53(self):
             toggles.USH_CASE_CLAIM_UPDATES.enabled(self.domain)
             and self._require_minimum_version('2.53')
         )
+
+    @property
+    def supports_ush_empty_case_list_text_2_54(self):
```
Comment (snopoke) [inline corehq/apps/app_manager/feature_support.py:213]: nit: rename
```suggestion
    def supports_empty_case_list_text(self):
```

Comment (Jtang-1) [reply corehq/apps/app_manager/feature_support.py]: [d10271a](https://github.com/dimagi/commcare-hq/pull/32552/commits/d10271a1a3976f72058a70a2a56843cc2fdb1b9d)

**Comment by Jtang-1** [comment]
> If so, a solution I have is to pass back nothing from formplayer if there's no customized text. And on the template, check if `noItemsText` is empty, and if so, display `{% trans "List is empty." %}`. I wasn't able to find a pattern we use to translate default text passed through a variable (even though i'm sure we do it).

Note: I ended up implementing this. But to have `noItemsText` passed as `null` from formplayer, then doing a null check on web app template to display `{% trans "List is empty." %}` if it is null. So that will be displayed when the FF is off or when commcare version < 2.54. 

[Discussion](https://github.com/dimagi/formplayer/pull/1320#discussion_r1091682558) I had with Shubham on FP PR

File: corehq/apps/app_manager/feature_support.py [hunk 2]
```diff
@@ -208,3 +208,7 @@ def ush_case_claim_2_53(self):
...
+        return self._require_minimum_version('2.54')
```
Comment (stephherbers) [inline corehq/apps/app_manager/feature_support.py]: Usually here is a nice place to also check for FF so it doesn't have to be a separate check, is that something you would consider?

File: corehq/apps/app_manager/feature_support.py [hunk 3]
```diff
@@ -211,4 +211,7 @@ def ush_case_claim_2_53(self):
 
     @property
     def supports_ush_empty_case_list_text_2_54(self):
-        return self._require_minimum_version('2.54')
+        return (
```
Comment (orangejenny) [inline corehq/apps/app_manager/feature_support.py:214]: nice change

File: corehq/apps/app_manager/feature_support.py (hunk 2 above)
Comment (stephherbers) [reply corehq/apps/app_manager/feature_support.py]: ha I see this change later on in the PR-- fantastic

**Review by stephherbers — APPROVED** [review]
Thank you for the modifications and tests

**Pull request merged**